src/Flight/FlightService.java

```diff
@@ -1,6 +1,143 @@
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
 // TODO: Implement the Service Layer, this contains all Business Logic (anything that makes sense in the scope of a business)
 public class FlightService {
+    private Flight[] flights = new Flight[5];
+
+
+    public Flight[] getFlightInfo(){
+        if (isEmpty(flights)) {
+            System.out.println("No flight info");
+            return null;
+        } else {
+            System.out.println("Returning flight information....");
+            return flights;
+        }
+    }
+
+    public void createFlight(Flight flight){
+        int indexToReplace = getFirstNull(flights);
+        if (indexToReplace == -1) {
+            System.out.println("Sorry, our flight database is full, please try again later");
+        } else if(!validateMinFlight(flight)){
+            System.out.println("Invalid information provided.");
+        } else {
+            flights[indexToReplace] = flight;
+            System.out.printf("Flight %s successfully added\n", flight);
+        }
+    }
+
+    public Flight findById(int flightNumber){
+        for (Flight flight : flights) {
+            if (flight != null && flight.flightNumber == flightNumber) {
+                return flight;
+            }
+        }
+
+        return null;
+    }
+
+
+    public boolean updateFlight(Flight flightToUpdate, String timeArrival, String timeDeparture, int pilot, int airline) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+
+        flightToUpdate.setTimeArrival(LocalDateTime.parse(timeArrival, formatter));
+        flightToUpdate.setTimeDeparture(LocalDateTime.parse(timeDeparture, formatter));
+        flightToUpdate.setPilot(pilot);
+        flightToUpdate.setAirline(airline);
+
+        if(!validateFullFlight(flightToUpdate)) return false;
+
+        return true;
+    }
+
+
+    private boolean isEmpty(Flight[] arr) { // defining the parameter of a string array to be included when executing this mehtod
+        for (Flight element : arr) { // enhanced for each loop, that iterates through the arrays elements returning & assigning the value to the declared variable flight
+            if (element != null) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public int getFirstNull(Flight[] arr) { //get the first null index to replace, otherwise return -1
+        for (int i = 0; i < arr.length; i++) {
+            if (arr[i] == null) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    private boolean validateMinFlight(Flight flight) {
+        if (flight == null) {
+            return false;
+        }
+
+        if (flight.flightNumber < 1000 || flight.flightNumber > 999999) {
+            System.out.println("Invalid flight number");
+            return false;
+        }
+
+        if (flight.originAirport.length() != 3 || flight.destinationAirport.length() != 3) {
```

Comment:
Check for casing here? Then either throw an error or just fix the casing. I think there is a method that automatically does this.